jiant/proj/main/modeling/taskmodels.py

```diff
@@ -276,14 +282,94 @@ def get_output_from_encoder(encoder, input_ids, segment_ids, input_mask) -> Enco
         EncoderOutput containing pooled and unpooled model outputs as well as any other outputs.
 
     """
+    model_arch = ModelArchitectures.from_encoder(encoder)
+    if model_arch in [
+        ModelArchitectures.BERT,
+        ModelArchitectures.ROBERTA,
+        ModelArchitectures.ALBERT,
+        ModelArchitectures.XLM_ROBERTA,
+    ]:
+        pooled, unpooled, other = get_output_from_standard_transformer_models(
+            encoder=encoder, input_ids=input_ids, segment_ids=segment_ids, input_mask=input_mask,
+        )
+    elif model_arch in [
+        ModelArchitectures.XLM,
+    ]:
+        pooled, unpooled, other = get_output_from_xlm_with_lang_handing(
+            encoder=encoder, input_ids=input_ids, input_mask=input_mask, batch=batch,
+        )
+    elif model_arch in [
+        ModelArchitectures.BART,
+        ModelArchitectures.MBART,
+    ]:
+        pooled, unpooled, other = get_output_from_bart_models(
+            encoder=encoder, input_ids=input_ids, input_mask=input_mask,
+        )
+    else:
+        raise KeyError(model_arch)
```

Comment:
I am still thinking about this but my preference here is to try and reduce the duplication of code in the `get_output_from_*` functions.

Reply:
IMO we should save that for a later refactor. (I'm thinking something along the lines of a corresponding wrapper around the encoder model that exposes those signatures, rather than standalone functions.)

Reply:
This would also the stuff here: https://github.com/jiant-dev/jiant/blob/dcb58be06a0187297a4dca741942d191fdaaa182/jiant/proj/main/modeling/model_setup.py#L266-L286